xla/stream_executor/rocm/rocm_blas.cc

```diff
@@ -692,17 +692,15 @@ bool ROCMBlas::GetBlasGemmAlgorithms(
   auto blas_lambda = [this, out_algorithms](auto handle, auto &&blas_func,
                                             auto &&...rest) {
     rocblas_int num_sols = 0;
-
-    if (auto ret = blas_func(handle, std::forward<decltype(rest)>(rest)...,
-                             nullptr, &num_sols);
-        ret != rocblas_status_success) {
-      return ret;
-    }
-    solutions_.resize(num_sols);
+    // If get_solutions call fails, we still can use the default (fallback) 
+    // algorithm which is available for almost all number types.
     if (auto ret = blas_func(handle, std::forward<decltype(rest)>(rest)...,
-                             solutions_.data(), &num_sols);
-        ret != rocblas_status_success) {
-      return ret;
+                         nullptr, &num_sols); ret == rocblas_status_success) {
+      solutions_.resize(num_sols);
+      if(ret = blas_func(handle, std::forward<decltype(rest)>(rest)...,
```

Comment:
Linter will complain about the lack of space between `if` and `(ret`.

Can you add a space there?

Reply:
yep, fixed it